vault/activity_log.go

```diff
@@ -2009,6 +2046,195 @@ func processClientRecord(e *activity.EntityRecord, byNamespace summaryByNamespac
 	byMonth.add(e, startTime)
 }
 
+// handleEntitySegment processes the record and adds it to the correct month/namespace breakdown maps, as well as to the hyperloglog
+func (a *ActivityLog) handleEntitySegment(l *activity.EntityActivityLog, segmentTime time.Time, hll *hyperloglog.Sketch, opts pqOptions) error {
```

Comment:
Does it make sense to have this as a receiver method? AFAICT, we're not doing anything with the `activityLog` here and below.

Reply:
Take it with a grain of salt, but in my opinion the code reads better the way it's currently written than if this method wasn't a receiver.